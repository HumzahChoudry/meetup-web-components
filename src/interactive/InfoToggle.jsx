import PropTypes from 'prop-types';
import React from 'react';

import Button from "../forms/Button";
import Tooltip from "./Tooltip";

const InfoTooltipTrigger = () => (
	<Button reset>
		<span className="infoToggle-trigger align--center" role="img">
			?
		</span>
	</Button>
);

const InfoToggle = ({
	className,
	label,
	tooltipId,
	tooltipProps,
	tooltipContent,
	...other
}) => {

	return (
		<div
			className={className}
			{...other}
		>
			<span>{label}</span>
			<Tooltip
				id={tooltipId}
				trigger={<InfoTooltipTrigger />}
				content={tooltipContent}
				{...tooltipProps}
			/>
		</div>
	);
};

InfoToggle.defaultProps = {
	tooltipProps: {
		align: 'right'
	}
};

InfoToggle.propTypes = {
	tooltipContent: PropTypes.element,
	label: PropTypes.string,
	tooltipId: PropTypes.string.isRequired,
	tooltipProps: PropTypes.object,
};

export default InfoToggle;
