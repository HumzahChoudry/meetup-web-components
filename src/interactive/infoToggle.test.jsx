import React from 'react';
import { shallow } from 'enzyme';
import InfoToggle from './InfoToggle';

const wrapper = shallow(<InfoToggle label="Get info" tooltipId="tooltipTestId" />);

describe('InfoToggle', () => {
	it('should render', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
