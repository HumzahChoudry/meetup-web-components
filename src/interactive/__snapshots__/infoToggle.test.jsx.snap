// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoToggle should render 1`] = `
<div>
  <span>
    Get info
  </span>
  <Tooltip
    align="right"
    direction="bottom"
    id="tooltipTestId"
    minWidth="0px"
    noPortal={false}
    trigger={<InfoTooltipTrigger />}
  />
</div>
`;
